subsys/bluetooth/mesh/proxy_cli.c

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2021 Xiaomi Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/byteorder.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/mesh.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_PROXY)
+#define LOG_MODULE_NAME bt_mesh_proxy_client
+#include "common/log.h"
+
+#include "mesh.h"
+#include "adv.h"
+#include "net.h"
+#include "rpl.h"
+#include "transport.h"
+#include "host/ecc.h"
+#include "prov.h"
+#include "beacon.h"
+#include "foundation.h"
+#include "access.h"
+#include "proxy.h"
+#include "gatt_cli.h"
+#include "proxy_msg.h"
+
+static struct bt_mesh_proxy_server {
+	struct bt_mesh_proxy_role *srv;
+	bool link_opened;
+	uint16_t net_idx;
+} servers[CONFIG_BT_MAX_CONN] = {
+	[0 ... (CONFIG_BT_MAX_CONN - 1)] = {
+		.net_idx = BT_MESH_KEY_UNUSED,
+	},
+};
+
+static bool allow_all_subnet;
+
+static struct bt_mesh_proxy_server *find_proxy_srv(uint16_t net_idx,
+						   bool conn, bool disconn)
+{
+	for (int i = 0; i < ARRAY_SIZE(servers); i++) {
+		if (!servers[i].srv) {
+			if (!disconn) {
+				continue;
+			}
+		} else if (!conn) {
+			continue;
+		}
+
+		if (servers[i].net_idx == net_idx) {
+			return &servers[i];
+		}
+	}
+
+	return NULL;
```

Comment:
Seems like `bt_conn_index()` could be used here as well?

Reply:
Under the existing structure, we cannot guarantee that the order of net_idx and bt_conn_index is the same, and I do not recommend changing it here.

Reply:
Yes, you're right.